### PR TITLE
HPCC-21529 ECLCC - Make sure audience and message class is sensible

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
@ghalliday Please can you review the logging changes in the compiler

File: system/jlib/jdebug.cpp [hunk 1]
```diff
@@ -3075,7 +3075,7 @@ void printProcMap(const char *fn, bool printbody, bool printsummary, StringBuffe
                 else if (strcmp(perms,"r--p")==0)
                         t = segtype_const;
                 else {
-                    ERRLOG("%s - unknown perms",perms);
+                    OERRLOG("%s - unknown perms",perms);
```
Comment (ghalliday) [inline system/jlib/jdebug.cpp]: probably IERR.  This is unexpected permissions reported by the operating system, not something the operator can change.

File: system/jlib/jthread.cpp [hunk 2]
```diff
@@ -1322,7 +1322,7 @@ static void CheckAllowedProgram(const char *prog,const char *allowed)
         if (WildMatch(head.str(),list.item(i)))
             return;
     }
-    ERRLOG("Unauthorized pipe program(%s)",head.str());
+    OERRLOG("Unauthorized pipe program(%s)",head.str());
```
Comment (ghalliday) [inline system/jlib/jthread.cpp]: Possibly AERRLOG should this  be audited?  Jake what are the rules for audit messages?

File: system/jlib/jthread.cpp [hunk 3]
```diff
@@ -1444,7 +1444,7 @@ class CWindowsPipeProcess: implements IPipeProcess, public CInterface
             if (_title) {
                 StringBuffer errstr;
                 formatSystemError(errstr, GetLastError());
-                ERRLOG("%s: PIPE process '%s' failed: %s", title.get(), prog, errstr.str());
+                OERRLOG("%s: PIPE process '%s' failed: %s", title.get(), prog, errstr.str());
```
Comment (ghalliday) [inline system/jlib/jthread.cpp]: Jake O or U?

File: system/jlib/jthread.cpp [hunk 4]
```diff
@@ -1730,7 +1730,7 @@ static unsigned dowaitpid(HANDLE pid, int mode)
                 return WEXITSTATUS(stat);
             else if (WIFSIGNALED(stat))
             {
-                ERRLOG("Program was terminated by signal %u", (unsigned) WTERMSIG(stat));
+                OERRLOG("Program was terminated by signal %u", (unsigned) WTERMSIG(stat));
```
Comment (ghalliday) [inline system/jlib/jthread.cpp]: I think I rather than O

File: system/jlib/jthread.cpp [hunk 5]
```diff
@@ -1746,7 +1746,7 @@ static unsigned dowaitpid(HANDLE pid, int mode)
         if (err == ECHILD) 
             break;
         if (err!=EINTR) {
-            ERRLOG("dowait failed with errcode %d",err);
+            OERRLOG("dowait failed with errcode %d",err);
```
Comment (ghalliday) [inline system/jlib/jthread.cpp]: Also I think I rather than O

**Review by ghalliday — COMMENTED** [review]
@shamser a few comments on the 1st commit

**Comment by shamser** [comment]
@ghalliday

**Comment by shamser** [comment]
@ghalliday

File: ecl/hql/hqlutil.cpp [hunk 6]
```diff
@@ -7657,7 +7657,7 @@ bool ErrorSeverityMapper::addMapping(const char * category, const char * value)
 {
     if (!category || !*category)
     {
-        ERRLOG("Error: No warning category supplied");
+        IERRLOG("Error: No warning category supplied");
```
Comment (ghalliday) [inline ecl/hql/hqlutil.cpp]: probably U

File: ecl/hql/hqlutil.cpp [hunk 7]
```diff
@@ -7669,7 +7669,7 @@ bool ErrorSeverityMapper::addMapping(const char * category, const char * value)
     ErrorSeverity severity = getSeverity(action);
     if (severity == SeverityUnknown)
     {
-        ERRLOG("Error: Invalid warning severity '%s'", value);
+        IERRLOG("Error: Invalid warning severity '%s'", value);
```
Comment (ghalliday) [inline ecl/hql/hqlutil.cpp]: U again

File: ecl/hql/hqlutil.cpp [hunk 8]
```diff
@@ -7687,7 +7687,7 @@ bool ErrorSeverityMapper::addMapping(const char * category, const char * value)
         return true;
     }
 
-    ERRLOG("Error: Mapping doesn't specify a valid warning code or category '%s'", category);
+    IERRLOG("Error: Mapping doesn't specify a valid warning code or category '%s'", category);
```
Comment (ghalliday) [inline ecl/hql/hqlutil.cpp]: U

File: ecl/hqlcpp/hqlcpp.cpp [hunk 9]
```diff
@@ -6211,7 +6210,7 @@ void HqlCppTranslator::doBuildCall(BuildCtx & ctx, const CHqlBoundTarget * tgt,
 
         if (arg >= maxArg)
         {
-            PrintLog("Too many parameters passed to function '%s'", str(expr->queryName()));
+            UERRLOG("Too many parameters passed to function '%s'", str(expr->queryName()));
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlcpp.cpp]: I rather than U - should have been reported earlier if user error

File: ecl/hqlcpp/hqlcpp.cpp [hunk 10]
```diff
@@ -6348,7 +6347,7 @@ void HqlCppTranslator::doBuildCall(BuildCtx & ctx, const CHqlBoundTarget * tgt,
     if (arg < maxArg)
     {
         //MORE: Process default parameters...
-        PrintLog("Not enough parameters passed to function '%s'", str(expr->queryName()));
+        UERRLOG("Not enough parameters passed to function '%s'", str(expr->queryName()));
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlcpp.cpp]: I rather than U (as above)

**Review by ghalliday — COMMENTED** [review]
@shamser a few comments

**Comment by shamser** [comment]
@ghalliday

**Comment by ghalliday** [comment]
@shamser please can you squash

**Comment by shamser** [comment]
@ghalliday Squashed.

**Pull request merged**